aptos-move/aptos-vm/src/move_vm_ext/write_op_converter.rs

```diff
@@ -189,8 +189,10 @@ impl<'r> WriteOpConverter<'r> {
         };
         Ok(GroupWrite::new(
             self.convert(state_value_metadata_result, metadata_op, false)?,
+            // TODO[agg_v2](fix): Converting the inner ops from Vec to BTreeMap. Try to have
+            // uniform datastructure to represent the inner ops.
+            inner_ops.into_iter().collect(),
```

Comment:
Why? inner ops are already a map, why convert to vector (given that in constructor we then have to convert back?)

Reply:
There is a comment/TODO above for fixing it :)